pkg/prowgen/prowgen.go

```diff
@@ -183,10 +207,28 @@ func hashDailyCron(job string) string {
 	return fmt.Sprintf("%d %d * * *", minute, hour)
 }
 
-func GeneratePeriodicForTest(jobBaseBuilder *prowJobBaseBuilder, info *ProwgenInfo, cron string, interval string, releaseController bool, pathAlias *string) *prowconfig.Periodic {
-	// Periodics are rehearsable
+type GeneratePeriodicOptions struct {
+	releaseController bool
+	pathAlias         *string
+	disableRehearsal  bool
+}
+
+type GeneratePeriodicOption func(options *GeneratePeriodicOptions)
+
+func FromConfigSpec(configSpec *cioperatorapi.ReleaseBuildConfiguration) GeneratePeriodicOption {
+	return func(options *GeneratePeriodicOptions) {
+		options.pathAlias = configSpec.CanonicalGoRepository
+	}
+}
+
+func GeneratePeriodicForTest(jobBaseBuilder *prowJobBaseBuilder, info *ProwgenInfo, cron string, interval string, options ...GeneratePeriodicOption) *prowconfig.Periodic {
```

Comment:
`cron` and `interval` should be in the options as well.

Reply:
My thinking there was that one of them must be provided, so they aren't really optional. I can move them though if you think thats best.